test/kitchen_sink/struct_test.exs

```diff
@@ -27,5 +27,20 @@ defmodule KitchenSink.StructTest do
       }
       assert Struct.to_map(input) == expected
     end
+
+    test "struct with list of primitives converts to map" do
+      input = %TestStructNested{
+        field1: %TestStruct{},
+        field2: ["something", "else"]
+      }
+      expected = %{
+        field1: %{
+          age: nil,
```

Comment:
@Corvillus & @olafura was there a decision around this, one that was maybe made in a live chat?